routes/sites.js

```diff
@@ -1,25 +1,84 @@
 'use strict';
 
-// const pool = require('../config/config').pool;
+const Promise = require('bluebird');
+const query = require('../lib/utils').query;
 
+/**
+ * Gets all sites.
+ *
+ * @param {Object} req The given request object
+ * @return {Promise} The promise
+ */
 function getSites(req) {
+  if (req.query && req.query.coach_id) {
+    return query('SELECT site_id, site_name, site_address FROM Acct NATURAL JOIN AcctToProgram NATURAL JOIN Program NATURAL JOIN Site WHERE acct_id = ' + req.query.coach_id);
+  }
 
+  return query('SELECT * FROM Site');
 }
 
+/**
+ * Creates a site.
+ *
+ * @param {Object} req The given request object
+ * @return {Promise} The promise
+ */
 function postSite(req) {
+  if (!req.body || !req.body.site_name || !req.body.site_address) {
+    return Promise.reject({
+      status: 406,
+      message: 'Must provide site\'s name, and address'
+    });
+  }
 
+  return query('INSERT INTO Site (site_name, site_address) VALUES ("' + req.body.site_name + '", "' + req.body.site_address + '")');
 }
 
+/**
+ * Get a sites with given id.
+ *
+ * @param {Object} req The given request object
+ * @return {Promise} The promise
+ */
 function getSite(req) {
-
+  return query('SELECT * FROM Site WHERE site_id = ' + req.params.site_id);
```

Comment:
I dmed you. But I think that you `isValidDate` and `isPositiveInteger` should be in `utils.js` so everyone can use them. So I will not make this change now as we can open a PR later using your functions in `utils.js`! :smile: